### PR TITLE
Splash UI Overhaul, Workflow & Linting Changes, Hotfix for Planets

### LINKED ISSUE #370: Login page buttons are ugly
Make them translucent rounded rectangles?

### PR DESCRIPTION
**Pull request by DamienVesper**
This PR overhauls the UI, switching from ugly rectangular buttons to translucent rounded rectangular ones, better fitting in with the rest of the UI theme for ingame controls. It also creates material design buttons for SFX instead of using inline spans.

The Lint workflow has been reformatted so that the job name can be read at a glance, and if one needed to know what tool was being used to lint the code, they could simply look at the build steps, which have the tool labeled in parentheses.

Additionally, linting has been changed to enforce 4-space indentation, to prevent the mix of tabs and spaces that had existed in the repository which lead to misinterpreted indentation.

This also provides a hotfix for a change to the planet ammo restock mechanism.

This pull request resolves #370.

image
image
image

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,16 +1,17 @@
-name: Node CI
+name: Lint
 
 on: [push,pull_request]
 
 jobs:
   lint-js:
-    name: Lint Javascript/Typescript via eslint
+    name: JS & TS
```
Comment (johnnyapol) [inline .github/workflows/lint.yml:7]: This is probably a question of personal choice - but I like having the actual program doing the enforcement visible. That way if a new contributor runs into issues they know what program to use to format immediately.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -24,11 +25,10 @@ jobs:
       env:
         CI: true
   lint-py:
-    name: Lint via python-black
+    name: Python
```
Comment (johnnyapol) [inline .github/workflows/lint.yml:28]: Same as above

File: server_src/player.js [hunk 3]
```diff
@@ -834,9 +834,10 @@ class Player {
         if (tick % 2 != 0 || squaredDist(p, this) > square(512)) return;
 
         // cooldown to prevent chat spam when 2 people are on the planet
-        const cool = p.cooldown;
-        if (cool < 0) {
+        let cool = p.cooldown;
```
Comment (johnnyapol) [inline server_src/player.js:837]: These don't look like style changes but rather a bugfix so I guess I'll let it slide

File: server_src/player.js [hunk 4]
```diff
@@ -1050,10 +1051,8 @@ class Player {
     }
 
     EMP (t) {
- 	    if (this.ship >= 16 && this.ship <= 20) t *= 1.25; // Emp works better on elite ships.
-        if (this.ship === 21 && this.health * 1.05 < this.maxHealth) this.health *= 1.05; // It will also heal the ship a very small bit.
-
-        if (this.shield) t *= 0.33; // Shield offers some protection for electronic components
```
Comment (johnnyapol) [inline server_src/player.js:1056]: Why did this go away? it's not documented in the change

Comment (tardisfromtornspace) [reply server_src/player.js:1056]: No idea, it was part of the EMP features that made Elite Ships weaker to that weapon, plus the reason why r21 didn't have a C-Slot on the first place.

Comment (tardisfromtornspace) [reply server_src/player.js:1056]: this is fixed in lb-changes though

Comment (DamienVesper) [reply server_src/player.js:1056]: Reverted back to original.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (DamienVesper) [reply .github/workflows/lint.yml:7]: One can open up the build to see what tool is being used to lint the code. If they would like to know what this tool is, they would most likely be looking at the entire workflow anyways, not just the title. What was previously done normally went offscreen and required someone to actually open up the workflow in a separate tab to read it.

README should probably be updated for this as a better replacement instead, as that's where contributors would first go to get their questions answered.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (DamienVesper) [reply .github/workflows/lint.yml:28]: See above.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (johnnyapol) [reply .github/workflows/lint.yml:7]: I mean it renders fine for me (and this is having the tab in a half-window) 
image

There's also the other topic that changing the names of these things requires changing the protection hooks which normally wouldnt be annoying but this has happened several times now

Comment (DamienVesper) [reply .github/workflows/lint.yml:7]: image

Comment (DamienVesper) [reply .github/workflows/lint.yml:7]: That's my issue, if one wants to know at a glance their workflows status, it shouldn't go offscreen. But if they want to know more about the build sequence, then they should normally hit the individual 'Details' button to dive further in-depth about it.

**Pull request merged**